Program.cs

```diff
@@ -993,16 +993,62 @@ internal bool End()
                     return true;
 
                 default:
-                    TypeWriter.WriteLine("bye for now :)",TypeWriter.Speed.Talk);
+                    TypeWriter.WriteLine("bye for now :)", TypeWriter.Speed.Talk);
+                    TypeWriter.WriteLine("");
+                    UpdateLeaderBoard(totalXP);
                     return false;
             }
         }
+
+        private void UpdateLeaderBoard(int totalXP)
+        {
+            //read the leaderboard
+            var rawLeaderBoard = new List<string>(File.ReadAllLines(@".\Data\LeaderBoard.txt"));
+            var leaderBoard = new List<LeaderBoardEntry>();
+            
+            //add to the leaderboard
+            var currentGameEntry = new LeaderBoardEntry(userName, totalXP);
+            leaderBoard.Add( currentGameEntry);
+
+            //fill the leaderboard will the raw leaderboard
+            foreach( var rawEntry in rawLeaderBoard)
+            {
+                leaderBoard.Add( new LeaderBoardEntry( rawEntry ) );
+            }
+            
+            leaderBoard.Sort();
+            leaderBoard.Reverse();
+
+            if (leaderBoard.Count > 9)
```

Comment:
If we think it can be greater than 10 .. then do we need to remove more than just item 10 ?